pkg/volume/csi/csi_attacher.go

```diff
@@ -390,6 +390,14 @@ func (c *csiAttacher) MountDevice(spec *volume.Spec, devicePath string, deviceMo
 		csiSource.VolumeAttributes,
 		mountOptions)
 
+	defer func() {
+		if err != nil && stageUnstageSet{
```

Comment:
if NodeStageVolume fails (fetal failures), it seems there is no need to call NodeUnstageVolume. But in the case of timeout failures for NodeStageVolume, we need to unstage. There is another PR trying to address that https://github.com/kubernetes/kubernetes/pull/82492